File: python/p008.py
# Largest product in a series

import os
import numpy as np
from itertools import chain


def solve():
    data = np.array(
        [
            *chain.from_iterable(
                np.loadtxt(os.path.join("..", "data", "p008.txt"), dtype=str))
        ],
        dtype=np.uint8,
    )

    return max(
        np.prod(data[i:(i + 13)], dtype=np.uint64) for i in range(1000 - 13))


if __name__ == "__main__":
    print(solve())

File: julia/p008.jl
# Largest product in a series

function solve()
    open(joinpath("..", "data", "p008.txt")) do data
        number = join(readlines(data))
        return maximum(
            prod(parse(Int, n) for n in (number[i:(i + 12)]))
            for i in 1:(1000 - 13))
    end
end


print(Int(solve()))
